dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt

```diff
@@ -175,18 +230,83 @@ internal object CoreFeature {
         systemInfoProvider.register(appContext)
 
         // Network Info Provider
+        setupNetworkInfoProviders(appContext, ndkCrashDirectory, ndkCrashDirectoryTemp)
+
+        // User Info Provider
+        userInfoProvider = DatadogUserInfoProvider(
+            DefaultConsentAwareDataWriter(
+                trackingConsentProvider,
+                DataProcessorFactory(
+                    NdkFileOrchestrator(
+                        File(
+                            ndkCrashDirectoryTemp,
+                            DatadogNdkCrashHandler.LAST_USER_INFORMATION_FILE_NAME
+                        )
+                    ),
+                    NdkFileOrchestrator(
+                        File(
+                            ndkCrashDirectory,
+                            DatadogNdkCrashHandler.LAST_USER_INFORMATION_FILE_NAME
+                        )
+                    ),
+                    UserInfoSerializer(),
+                    "",
+                    uploadExecutorService,
+                    NoOpEventMapper()
+                ) { fileOrchestrator, eventSerializer, _ ->
+                    NdkFileWriter(fileOrchestrator, eventSerializer)
+                },
+                DefaultMigratorFactory(
+                    ndkCrashDirectoryTemp.absolutePath,
+                    ndkCrashDirectory.absolutePath,
+                    uploadExecutorService
+                )
+            )
```

Comment:
This part is still a bit huge, but I guess we need to have it somewhere. There might be some readability enhancement nevertheless. 
- This `DefaultConsentAwareDataWriter` uses almost exactly the same config as the one below for the Network info. It might be worth creating a method/class to factorize the code?
- Maybe you could create an `NdkPersistenceStrategy` class to setup this, which could also be reused in the `DatadogNdkCrashHandler`  to know which file(s) should be read from, that way getting more consistent in the way we perform File I/O in the SDK

Reply:
I need to check, the way I see it now if we want to go with the persistence strategy I am afraid there will be 2 classes (one for NetworkInfo and one for UserInfo) as the storage location (orchestrator is different). I would rather keep it like this but try to extract more code from there in private methods.